src/lib.rs

```diff
@@ -85,6 +88,35 @@ impl Supercluster {
             stride: 6,
             points: vec![],
             cluster_props: vec![],
+            use_non_geospatial_coords: false,
+        }
+    }
+
+    /// Create a new instance of `Supercluster` with the specified configuration settings.
+    ///
+    /// This should work with non-geospatial points and will not perform any mercator transform on
+    /// the data.
+    ///
+    /// # Arguments
+    ///
+    /// - `options`: The configuration options for Supercluster.
+    ///
+    /// # Returns
+    ///
+    /// A new `Supercluster` instance with the given configuration.
+    pub fn new_non_geospatial(options: Options) -> Self {
```

Comment:
I don't think we need a new method that is exactly the same except for the `use_non_geospatial_coords` property value. We can reuse the existing `new` by setting the `use_non_geospatial_coords` property from the `options` parameter.

Reply:
Can you advise how to do this? I am fairly new to Rust, and I cannot think of a way to add a property to `Options` without breaking backwards compatibility.

Reply:
Sure, add a new boolean flag here: https://github.com/chargetrip/supercluster-rs/blob/main/src/lib.rs#L27.
After that, you will have a property value to set it here: https://github.com/chargetrip/supercluster-rs/blob/main/src/lib.rs#L87.

Reply:
That does not sound right. If let's say I add a new parameter `Options` `is_non_geospatial: bool`, then any place that I set the `Options` that are passed to `Supercluster` will need to include `is_non_geospatial` in those options.

Reply:
What about using an enum instead of a boolean flag? This way it is clearer which cluster type we want to use. Check the comments above about this suggestion.

Reply:
Indeed, we can use an enum type to control it instead of a boolean flag.
The only place we need options is when we initialise `Supercluster`.

Reply:
Ok.  So, I add:
```
cluster_type: ClusterType
```

to `Options`.

Now any previous code that instantiates a new `Supercluster` that wants to use the default geo-spatial use-case will break because `cluster_type` is missing.

Reply:
Correct, we introduce a breaking change but I will mention it in the upcoming `Changelog.md`.

Reply:
Ok, as long as you are cool with it, I like this a lot more.

I was mostly trying to protect against a breaking change.

In other languages (i.e. Python), I would just introduce the new option with a default value. I wish there was some way to do something similar in Rust. Alas, Rust demands being explicit. Not a bad thing, but just different.

Ok, I will introduce the breaking change as suggested!

Reply:
thanks!
I will update CI to generate release notes and it will be visible that we generated a breaking change.